modules/systemd/src/main/java/org/elasticsearch/systemd/SystemdPlugin.java

```diff
@@ -77,19 +83,23 @@ public Collection<?> createComponents(PluginServices services) {
          * Therefore, every fifteen seconds we send systemd a message via sd_notify to extend the timeout by thirty seconds. We will cancel
          * this scheduled task after we successfully notify systemd that we are ready.
          */
-        extender.set(services.threadPool().scheduleWithFixedDelay(() -> {
-            final int rc = sd_notify(0, "EXTEND_TIMEOUT_USEC=30000000");
-            if (rc < 0) {
-                logger.warn("extending startup timeout via sd_notify failed with [{}]", rc);
-            }
-        }, TimeValue.timeValueSeconds(15), EsExecutors.DIRECT_EXECUTOR_SERVICE));
+        extender.set(
+            services.threadPool()
+                .scheduleWithFixedDelay(
+                    () -> { systemd.notify_extend_timeout(30); },
+                    TimeValue.timeValueSeconds(15),
+                    EsExecutors.DIRECT_EXECUTOR_SERVICE
+                )
+        );
         return List.of();
     }
 
-    int sd_notify(@SuppressWarnings("SameParameterValue") final int unset_environment, final String state) {
-        final int rc = Libsystemd.sd_notify(unset_environment, state);
-        logger.trace("sd_notify({}, {}) returned [{}]", unset_environment, state, rc);
-        return rc;
+    void notifyReady() {
+        systemd.notify_ready();
```

Comment:
These methods don't guard against `systemd` being null, which it could be if a noop/abstract implementation is used